src/pymatgen/core/operations.py

```diff
@@ -243,12 +244,16 @@ def translation_vector(self) -> np.ndarray:
     @property
     def inverse(self) -> Self:
         """Inverse of transformation."""
-        inverse = np.linalg.inv(self.affine_matrix)
-        return type(self)(inverse)
+        new_instance = copy.deepcopy(self)
```

Comment:
strictly speaking we don't need to make a deep copy here as the only mutable attribute is overwritten by `np.linalg.inv(self.affine_matrix)` which would generate a new instance anyway, but I still decide to make deep copy to avoid potential breakage should any mutable attribute be added later.